engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/ExecutionStore.scala

```diff
@@ -124,18 +130,22 @@ final case class ExecutionStore(private val statusStore: Map[JobKey, ExecutionSt
   }
 
   private def arePrerequisitesDone(key: JobKey): Boolean = {
-    def calculateUpstreamDone(upstream: GraphNode, index: ExecutionIndex) = upstream match {
-        // FIXME this won't hold conditionals in scatters for example
+    def calculateUpstreamDone(upstream: GraphNode, index: ExecutionIndex): Boolean = upstream match {
         // OuterGraphInputNode signals that an input comes from outside the graph.
         // Depending on whether or not this input is outside of a scatter graph will change the index which we need to look at
-      case outerNode: OuterGraphInputNode => doneKeys.contains(outerNode, None)
+      case outerNode: ScatterVariableNode =>
+        calculateUpstreamDone(outerNode.linkToOuterGraph.graphNode, None)
+      case outerNode: OuterGraphInputNode =>
+        calculateUpstreamDone(outerNode.linkToOuterGraph.graphNode, index)
       case _: CallNode | _: ScatterNode | _: ExpressionNode | _: ConditionalNode => doneKeys.contains(upstream, index)
       case _ => true
     }
 
     def upstreamAreDone = key.node.upstreamPorts forall {
       // The collector is at index None, so if this is a scatter gather port ignore the key index
       case upstreamPort: ScatterGathererPort => calculateUpstreamDone(upstreamPort.executionNode, None)
+      case upstreamPort: ConditionalOutputPort =>
+        statusStore.get(ConditionalCollectorKey(upstreamPort, key.index, upstreamPort.g(()))).exists(_.isDoneOrBypassed)
```

Comment:
Re-creating a fake key just to check that one with the same hash is in the store feels kind of weird.
I think you could use the `doneKeys` table that contains all the keys in `DoneOrBypassed` status, grouped by node and index and check that the node is in there instead

Reply:
This is because `doneKeys` is based on GraphNode, whereas I need to check the `ConditionalCollector` (which doesn't have it's own Node)

`Collector` != `ConditionalNode`

Reply:
please comment this